cmd/sonobuoy/app/results.go

```diff
@@ -157,6 +174,62 @@ func result(input resultsInput) error {
 	return lastErr
 }
 
+// printHealthSummary pretends to work like printSinglePlugin
+// but for a "fake" plugin that prints health information
+func printHealthSummary(input resultsInput, r *results.Reader) error {
+	var err error
+
+	//For detailed view we can just dump the contents of the clusterHealthSummaryPluginName file
+	if input.mode == resultModeDetailed {
+		err = r.WalkFiles(func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if path == discovery.ClusterHealthFilePath() {
+				reader, ok := info.Sys().(io.Reader)
+				if !ok {
+					return errors.New("info.Sys() is not a reader")
+				}
+				_, err = io.Copy(os.Stdout, reader)
+				if err != nil {
+					return err
+				}
+			}
+			return nil
+		})
+		return err
+	}
+
+
+	// For summary and dump views, get the item as an object to iterate over.
+	clusterHealthSummary := discovery.ClusterSummary{}
+
+	err = r.WalkFiles(func(path string, info os.FileInfo, err error) error {
+		return results.ExtractFileIntoStruct(discovery.ClusterHealthFilePath(), path, info, &clusterHealthSummary)
```

Comment:
2 things about this section:
 - the walk files is unnecessary; `r` has a a `.FileReader(<file>)` method that will give you the reader you can copy from
 - Can we move the `discovery.ClusterHealthFilePath()` to be defined on the reader? When looking at the previous point, I found there are similar methods named for important files in the tarball

Reply:
Oops; wrong 'Walk' section. I meant the one above this where you find the file by name yourself.

Reply:
Regarding `ClusterHealthFilePath`:
in the reader I see the other methods there are `Reader`'s methods, should `ClusterHealthFilePath` be a `Reader`'s method too ?
I am asking because then I'd need to instantiate a Reader in `pkg/discovery/summary.go` just to get the file name:
is it Ok if I just move it to the `reader.go` file, together with the `ClusterHealthFile` const, but leave it as a "standalone" func  instead ?

Reply:
This was an earlier comment; in light of the other discussion about how that reader stuff should work on a directory/tarball we can just leave this as-is for now and include it in the cleanup work